### PR TITLE
FIX: Update custom KMS Policy to add IAM Identities

### PR DESCRIPTION
**Pull request by dibyadhar**
#### 📲 What

To conditionally inject IAM Identities to the KMS Key that used for secret encryption within the cluster. If no IAM role is passed the KMS Policy has only access to the identity that is used to deploy the infrastructure.

#### 🤔 Why

This gives flexibility to use IAM User in lower environment and IAM Role in higher environment to access the cluster encryption key and infrastructure deployment.

There are few gotchas like currently apart from the identity used to deploy the infra nobody has access even to view the KMS key making it difficult to manage, this is fixed in later version of the terraform module (V 20.0.0 )

https://github.com/terraform-aws-modules/terraform-aws-eks/issues/2816

#### 🛠 How

Using conditional statement.

#### 👀 Evidence
image

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR.

#### 🕵️ How to test

Notes on how a reviewer can test the changes, e.g. how to run the tests.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

**Review by ravisinghkr — APPROVED** [review]

File: aws/modules/infrastructure_modules/eks/variables.tf [hunk 1]
```diff
@@ -139,3 +139,15 @@ variable "image_gc_low_threshold_percent" {
...
+  description = "[ Warn: breaking-change ] Making this value false will allow passing a custom KMS key via the provider_key_arn configuration"
+  type        = bool
+  default     = true
+}
+
+variable "kms_key_administrators" {
+  type        = list(string)
+  description = "List of IAM identities for the keys which is used to encrypt data within the Cluster"
+  default     = []
+}
```
Comment (ElvenSpellmaker) [inline aws/modules/infrastructure_modules/eks/variables.tf:153]: newline please

File: aws/modules/infrastructure_modules/eks/variables.tf [hunk 2]
```diff
@@ -139,3 +139,15 @@ variable "image_gc_low_threshold_percent" {
...
   default     = 80
 }
+
+variable "create_kms_key" {
+  description = "[ Warn: breaking-change ] Making this value false will allow passing a custom KMS key via the provider_key_arn configuration"
+  type        = bool
+  default     = true
+}
+
+variable "kms_key_administrators" {
```
Comment (ElvenSpellmaker) [inline aws/modules/infrastructure_modules/eks/variables.tf:149]: Is this variable used anywhere?

Comment (dibyadhar) [reply aws/modules/infrastructure_modules/eks/variables.tf:150]: Hi Jack
This is used in EKS Cluster Config
This enables to pass IAM Role to the policy which will be used for deployment where as in default scenario it falls back to the user used in IAC deployments.

File: aws/modules/infrastructure_modules/eks/variables.tf (hunk 1 above)
Comment (dibyadhar) [reply aws/modules/infrastructure_modules/eks/variables.tf]: added now, thanks.

File: aws/modules/infrastructure_modules/eks/variables.tf (hunk 2 above)
Comment (ElvenSpellmaker) [reply aws/modules/infrastructure_modules/eks/variables.tf:150]: I see you've added the variable `` here but it's not being used anywhere in the PR?

Comment (dibyadhar) [reply aws/modules/infrastructure_modules/eks/variables.tf:150]: Yeah sorry will take it out, doesn't require to be defined here.

**Pull request merged**